### PR TITLE
docu: more small release lead role updates

### PR DESCRIPTION
**Pull request by tpepper**
Fixed the embedded html links within the table, and added more words on
release themes.

Signed-off-by: Tim Pepper <tpepper@vmware.com>

### CODE REVIEW

File: release-team/role-handbooks/release-team-lead/README.md [hunk 1]
```diff
@@ -94,7 +94,13 @@ A guiding principle for the lead is be an arbiter of decisions, and not the prim
   <tr>
     <td>1</td>
     <td><p>- Start the release cycle
-<p>- Insure you are joined to the following Google Groups: [kubernetes-milestone-burndown](https://groups.google.com/forum/#!forum/kubernetes-milestone-burndown),[kubernetes-sig-release](https://groups.google.com/forum/#!forum/kubernetes-sig-release), [kubernetes-sig-leads](https://groups.google.com/forum/#!forum/kubernetes-sig-leads), [kubernetes-dev](https://groups.google.com/forum/#!forum/kubernetes-dev).
+<p>- Insure you are joined to the following Google Groups:
```
Comment (AishSundar) [inline release-team/role-handbooks/release-team-lead/README.md]: nit: s/insure/ensure

Comment (tpepper) [reply release-team/role-handbooks/release-team-lead/README.md]: fixed

**Pull request merged**